flair/models/lemmatization_model.py

```diff
@@ -0,0 +1,630 @@
+        self.longest_word_length = max(self.longest_word_length, target_lenght, max_lemma_lenght)
+
+        return embed_data, lemma_seq_list, mask, effective_data_lenght, max_lemma_lenght
+
+    def _item2seq(self, dictionary: Dictionary, data):
+        # Convert item into a sequence.
+        sequence = []
+        for item in data:
+            sequence.append(dictionary.get_idx_for_item(item))
+
+        return sequence
+
+    def _seq_zero_padding(self, seqs: list, target_length: int, fill_value=0):
+        for seq in seqs:
+            seq.extend([fill_value] * (target_length - len(seq)))
+        return seqs
+
+    def _binary_matrix(self, seqs: list):
+        matrix = []
+        for i, seq in enumerate(seqs):
+            matrix.append([])
+            for value in seq:
+                if value == 0:
+                    matrix[i].append(0)
+                else:
+                    matrix[i].append(1)
+        return matrix
+
+    def _get_state_dict(self):
+        model_state = {
+            "state_dict": self.state_dict(),
+            "hidden_size": self.hidden_size,
+            "embeddings": self.embeddings,
+            "character_dictionary": self.character_dictionary,
+            "n_layers": self.n_layers,
+            "tag_list": self.tag_list if "tag_list" in self.__dict__.keys() else None,
+            "tag_dictionary": self.tag_dictionary if "tag_dictionary" in self.__dict__.keys() else None,
+            "dropout": self.dropout,
+            "teacher_forcing_ratio": self.teacher_forcing_ratio,
+            "longest_word_length": self.longest_word_length,
+            "contextualized_embedding": self.contextualized_embedding
+        }
+        return model_state
+
+    @staticmethod
+    def _init_model_with_state_dict(state):
+        model = Lemmatization(
+            hidden_size=state["hidden_size"],
+            embeddings=state["embeddings"],
+            character_dictionary=state["character_dictionary"],
+            n_layers=state["n_layers"],
+            tag_list=state["tag_list"],
+            tag_dictionary=state["tag_dictionary"],
+            dropout=state["dropout"],
+            teacher_forcing_ratio=state["teacher_forcing_ratio"],
+            longest_word_length=state["longest_word_length"],
+            contextualized_embedding=state["contextualized_embedding"]
+        )
+        model.load_state_dict(state["state_dict"])
+        return model
+
+    @staticmethod
+    def _fetch_model(model_name) -> str:
+
+        google_drive_url = {
+            "multi": "https://drive.google.com/uc?id=1b1gHiOsqu2bovHlO0MZAeB8NWNNR88CJ",
+            "multi-with-pos": "https://drive.google.com/uc?id=1svkCI1HRGK-rD8AAWgNphGunNA62UEEv"}
+
+        if model_name in google_drive_url:
+            output = str(flair.cache_root) + "/models/lemma/"
+            model_path = gdown.download(google_drive_url[model_name], output, quiet=False)
```

Comment:
If the model already exists, it should not download again. Currently, it is downloading the model every time this method is called.